src/regex.cpp

```diff
@@ -191,7 +191,9 @@ BOOST_REGEX_DECL void BOOST_REGEX_CALL put_mem_block(void* p)
 
 #else
 
-#ifdef BOOST_HAS_THREADS
+#if defined(BOOST_REGEX_MEM_BLOCK_CACHE_LOCK_FREE)
+mem_block_cache block_cache;
```

Comment:
Yeah. that might be the simplest option for now. I will replace all boost::atomic with just std::atomic and static initialize that.